### DISCUSSION TITLE
Image pasting side by side no longer possible

### DISCUSSION THREAD

**Discussion by taustin81** [TriliumNext/Trilium#4476] [General]
It seems that the ability to paste images at a 50% (size) side by side is no longer possible with versions above v52 of Trilium. Will this capability be making a comeback or was this inadvertently overlooked with recent updates? -Or, am I missing something?

I'm currently, running Trilium through docker. My Trilium version is current, 0.62.2. I've searched through discussions for this topic however, perhaps I am not typing in the most accurate of description.

I believe I was originally running version 0.52.4 which has the ability to paste screen captures side by side, when both are set to 50%. I have tested this in 0.60.4 and 0.61.11 with the same results. 

I recall after pasting an image and scaling it to 50%, I could click next to the pasted image and see the cursor and paste my next image on the cursor location. Even if the new image is too large, once I scaled it to 50% it typically moved it next to the first image. This is no longer the behavior in Trilium. The cursor is not even visible any longer and the application does not even allow dragging images around. 

These are the steps I would typically perform:

1. I would paste a screen capture into a note. 
2. Click the resize button, 50% 
3. Click the button to shift the image to the left.
4. Click next to the image and see a cursor.
5. Paste another image on top of the cursor location.
6. The image would be pasted below due to its size, upon scaling it down to 50% the image would then shift next to the first image.
7. Optionally, with any pre-existing images, I could scale anything to 50% copy/paste a new one and resize the second one to 50% and Trilium would handle it as a side by side. I would imagine, we used to be able to also place four 25% scaled images in a row?

![image](https://github.com/zadam/trilium/assets/148650060/c9b4dfec-8512-4893-82af-bf18f6694479)
![image](https://github.com/zadam/trilium/assets/148650060/3aceed6a-b761-4815-bcaa-5e57801d9998)

Even with the current version 0.62.2 of Trilium, some of my existing notes I have, it will not allow for two images at 50% even though the images are exactly the same width. It seems it will only do a 25% and a 50%.
![image](https://github.com/zadam/trilium/assets/148650060/93e3ce1e-9dfd-4103-96e1-f03fabf80600)

[docker_trilium.txt](https://github.com/zadam/trilium/files/13467753/docker_trilium.txt)
[docker_version.txt](https://github.com/zadam/trilium/files/13467754/docker_version.txt)

**Comment by zadam** [comment]
Hello, image resizing/positioning changed together with how this feature evolved in CKEditor (which is used for editing text in Trilium). I can recollect some major changes in how images are handled / what are the options in the recent versions.

Could you please check if your desired behavior is possible with the CKEditor demo page? https://ckeditor.com/ckeditor-5/demo/feature-rich/

(If yes, then we might have something wrong in Trilium, and should be fixable)

**Reply by taustin81** [reply]
This is very interesting, yes! Although, one interesting caveat. I tried pasting some screenshots into the CKEditor and clicking next to the pasted image, I am not getting the cursor. So this is very much the behavior in Trilium I am experiencing. It does not allow me the possibility to place the cursor next to any pasted image unless those images were pasted in a previous version. Even dragging these two equally sized screen shots in CKEditor, its as if the're too wide, perhaps there is a padding size issue or something.

![image](https://github.com/zadam/trilium/assets/148650060/d62863b5-ec54-4d02-84d2-1b405cd43d9f)



It took me a minute to figure out CKEditor/table edit function. However, I managed to resize the table to 50% and I do see the cursor, clicking beside the table. Interestingly, clicking beside the table/image, I only see the cursor in the top three rows (beside the table itself), which seems rather odd. Nevertheless, I _see_ the cursor, which is not my present experience with Trillium and pasting images. Eventually after I begin to resize the second image, it does snap into place.

That placement, when the image is resized to an equal width of the first image is reminiscent of how Trilium originally behaved. Since the cursor placement was in an odd location in this table, I was able to manually drag the image into a vertical alignment. 
I can imagine a solid image  would not require additional dragging for vertical alignment (with an image of equal height) but snap into place once both images are equally sized. -Which is what I was expecting within Trilium.

![cke1](https://github.com/zadam/trilium/assets/148650060/243b2bab-e7d7-4f14-992c-276b0bb70cda)
![cke2](https://github.com/zadam/trilium/assets/148650060/c5317ab8-f95d-4d5e-81dc-fe456907022d)
![cke3](https://github.com/zadam/trilium/assets/148650060/7f687c2f-c283-48c2-b0ec-5bfde83e6259)
![cke4](https://github.com/zadam/trilium/assets/148650060/158026ea-969a-4550-a0b2-d1123b58f5fa)
![cke5](https://github.com/zadam/trilium/assets/148650060/770b99ff-8955-4059-9784-0aadc872daeb)
